src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -6,44 +6,138 @@
 
 namespace APIView.TreeToken
 {
+    
+    /// <summary>
+    /// Represents the type of a structured token.
+    /// All tokens should be content except for spacing tokens and url.
+    /// </summary>
     public enum StructuredTokenKind
     {
+        /// <summary>
+        /// Specifies that the token is content.
+        /// </summary>
         Content = 0,
+        /// <summary>
+        /// Space token indicating switch to new line.
+        /// </summary>
         LineBreak = 1,
+        /// <summary>
+        /// Regular single space.
+        /// </summary>
         NoneBreakingSpace = 2,
+        /// <summary>
+        /// 4 NonBreakingSpaces.
+        /// </summary>
         TabSpace = 3,
+        /// <summary>
+        /// Use this between method parameters. Depending on user setting this would result in a single space or new line.
+        /// </summary>
         ParameterSeparator = 4,
+        /// <summary>
+        /// A url token should have `LinkText` property i.e `token.Properties["LinkText"]` and the url/link should be the token value.
+        /// </summary>
         Url = 5
     }
 
     /// <summary>
-    /// Used to represent a APIView token its properties and tags for APIView parsers.
+    /// Represents an APIView token its properties and tags for APIView parsers.
     /// </summary>
 
     public class StructuredToken
     {
-        public HashSet<string> Tags
-        {
-            get { return TagsObj.Count > 0 ? TagsObj : null; }
-            set { TagsObj = value ?? new HashSet<string>(); }
-        }
+        /// <summary>
+        /// Token Id.
+        /// Previously known as DefinitionId.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Represents the type of a structured token.
+        /// </summary>
+        public StructuredTokenKind Kind { get; set; } = StructuredTokenKind.Content;
+        
+        /// <summary>
+        /// The token value which will be displayed. Spacing tokens don't need to have value.
+        /// </summary>
+        public string Value { get; set; } = string.Empty;
+        
+        /// <summary>
+        /// Properties of the token.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>GroupId: `doc` to group consecutive comment tokens as documentation.</description>
+        /// </item>
+        /// <item>
+        /// <description>NavigateToId: Id for navigating to where the node where the token is defined. Should match the definitionId of the referenced node.</description>
```

Comment:
@chidozieononiwu should this be `should match the Idof the referenced node` instead of `definitionId` ?